File: server/game_core/game_object.py
class GameObject(object):
    ID = None
    commands = None

    def __init__(self, id):
        self.ID = id
        self.commands = []

    def init(self, args):
        pass

    def update(self, delta):
        pass

    def act(self):
        pass

    def queue_command(self, command):
        self.commands.append(command)

File: src/client/lib/game/core/abstract/game_object.dart
import 'dart:collection';

import 'package:client/game/core/command.dart';

abstract class GameObject
{
  String ID;
  Queue<Command> commands;

  GameObject(this.ID)
  {
    commands = new Queue<Command>();
  }

  init([dynamic args=null]);

  update(num delta);

  act();

  queueCommand(Command command)
  {
    commands.add(command);
  }
}